mantidproject/mantid
# Fix unreliable tests: Disable ClearCache doc test

Description of work.

Uses a code block for the ClearCache example as the effect of the test caused new instrument definition files to be downloaded during the test. This had the effect that future pull requests could pick up an IDF from the user download area rather than the repository, as happen in #22175. 

Also, the buildscripts have been updated to remove the user configuration directory entirely
before running the tests.

**To test:**

Before making these changes I could get the tests to fail sometimes after running these steps (on Linux):

* `echo 3 | sudo tee /proc/sys/vm/drop_caches`
* `rm -fr ~/.mantid`
* `ctest --schedule-random --output-on-failure -j8 -R "MDAlgorithmsTest_ConvertCWPDMDToSpectraTest|AlgorithmsTest_MonitorEfficiencyCorUserTest|KernelTest_LogFilterTest|CurveFittingTest_DiffRotDiscreteCircleTest|APITest_FunctionAttributeTest|CurveFittingTest_MultiDomainFunctionTest|MDAlgorithmsTest_ForegroundModelTest|APITest_FunctionAttributeTest|AlgorithmsTest_MonitorEfficiencyCorUserTest|KernelTest_LogFilterTest"`

If it doesn't fail try repeating the 3 steps until it does.

Code review and make sure the tests pass.

Refs #22063 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
